rise-financial-es6.js

```diff
@@ -308,14 +311,32 @@
 
       this._goPending = false;
 
-      this._getData(
-        {
-          type: this.type,
-          duration: this.duration,
-        },
-        this._instruments,
-        this.instrumentFields
-      );
+      if ( this._initialGo ) {
+        this._initialGo = false;
+
+        // configure and execute initial request
+        this._getData(
+          {
+            type: this.type,
+            duration: this.duration,
+          },
+          this._instruments,
+          this.instrumentFields
+        );
+      } else {
+        // get cached data (if available)
+        this.$.data.getItem( this._getDataCacheKey(), ( cachedData ) => {
+          if ( this._refreshPending || !cachedData ) {
+            // refresh timer completed or there is no cached data available
```

Comment:
@stulees I feel as though some of the commenting here is extraneous. It's already pretty clear from the code and variable naming what is occurring. It may be good to use comments more judiciously in order to avoid some of the additional clutter that they add.